### DISCUSSION TITLE
How to log the SQL query that is built before running it?

### DISCUSSION THREAD

**Discussion by Blankeos** [porsager/postgres#932] [Q&A] [answered]
Let's say I have a query that looks like

```
sql`
  SELECT id, name FROM table
     ${whereClause}
     ${orderByClause}
;`
```

I have some variables replacing those above. How do I log the sql query that is built with this? (Not the result)

e.g.

SELECT id, name, FROM table WHERE name ilike 'awesome' ORDER BY age ASC

**Comment by K4leri** [comment]
Same problem. I have broken somewhere parameters that are inserted, but it is impossible to check without executing the query itself. And if it gives an error, it does not contain information about the SQL expression

**Comment by npham-dev** [comment, marked as answer]
I believe you can set debug to true in the config and inspect the query if it throws an error (not sure if there is a way to do this before the query is run):

```ts
postgres({ debug: true })

// later
try {
  await sql`faulty query ${dynamic and stuff}`
} catch(error) {
  console.log(error.query)
}
```

you can read more under the documentation, titled "Error handling".

**Reply by Blankeos** [reply]
Thanks @natmfat!
